WordPress/Classes/ViewRelated/Gutenberg/GutenbergSuggestionsViewController.swift

```diff
@@ -114,6 +114,13 @@ extension GutenbergSuggestionsViewController: UITextFieldDelegate {
             return true
         }
         let searchWord = nsString.replacingCharacters(in: range, with: string)
+
+        let dismissSequence = suggestionType.trigger + " "
+        guard searchWord != dismissSequence else {
+            onCompletion?(.success(""))
```

Comment:
This feels a bit hacky, so I'll explain what's being done here:
1. If the user types `"@ "`, the so-called "dismiss sequence", the user probably typed the space to dismiss the suggestions UI
2. So when this happens, we call the success handler and pass in an empty string. 

We pass in an empty string because on the RN side [here](https://github.com/WordPress/gutenberg/blob/c02903a4c7f964e96e27e0fae49df975f97e39ac/packages/rich-text/src/component/index.native.js#L502), the success handler inserts the `@`, followed by the username, followed by a space. Since we're essentially passing in an empty username, the result is that a `"@ "` is inserted into the post, which is the desired result.

Reply:
The same applies for cross-posts, with `"+ "` instead of `"@ "`.

Reply:
Thanks for the great explanation. Could we also add it to the code as a comment?
This will make it easier to understand in the future.

Reply:
Good idea! Added in 8f511be4520ff52aa8c5f1c690838a379265d7f5